library/Freckle/App/Http/Cache/Memcached.hs

```diff
@@ -0,0 +1,136 @@
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE OverloadedRecordDot #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+module Freckle.App.Http.Cache.Memcached
+  ( memcachedHttpCacheSettings
+  , memcachedHttpCodec
+  , memcachedHttpCache
+  ) where
+
+import Freckle.App.Prelude
+
+import Blammo.Logging (MonadLogger, logWarnNS)
+import Codec.Serialise (Serialise (..), deserialiseOrFail, serialise)
+import qualified Data.ByteString.Lazy as BSL
+import Data.String (IsString(..))
+import Database.Memcache.Types (Value)
+import Freckle.App.Http.Cache
+import Freckle.App.Memcached
+import qualified Freckle.App.Memcached.Client as Memcached
+import Freckle.App.OpenTelemetry (MonadTracer)
+import Network.HTTP.Client (Request, CookieJar, Cookie)
+import Network.HTTP.Types.Status (Status)
+import Data.CaseInsensitive (CI)
+import qualified Data.CaseInsensitive as CI
+import Network.HTTP.Types.Version (HttpVersion(..))
+import Network.HTTP.Types.Header (ResponseHeaders)
+import qualified Network.HTTP.Client.Internal as HTTP
+
+memcachedHttpCacheSettings
+  :: ( MonadUnliftIO m
+     , MonadLogger m
+     , MonadTracer m
+     , MonadReader env m
+     , HasMemcachedClient env
+     , Serialise body
+     )
+  => CacheTTL
+  -- ^ Default TTL, used when @max-age@ is not present
+  -> HttpCacheSettings m body Value
+memcachedHttpCacheSettings defaultTTL = HttpCacheSettings
+  { shared = False
```

Comment:
Should be `True`. This field used to be called `private` so I keep getting the meaning flipped in my head. Only a local Browser cache could really be considered un-shared.